plugins/repositories-plugin/src/main/java/org/pentaho/di/ui/repo/RepositoryConnectMenu.java

```diff
@@ -197,21 +200,22 @@ public void widgetSelected( SelectionEvent selectionEvent ) {
         connectionMenu.setVisible( true );
       }
 
-      private String truncateName( String name ) {
-        GC gc = new GC( toolBar );
-        Point size = gc.textExtent( name );
-        if ( size.x <= MAX_REPO_NAME_PIXEL_LENGTH ) { // repository name is small enough to fit just return it.
-          gc.dispose();
-          return name;
-        }
-        String originalName = name;
-        while ( gc.textExtent( name + "..." ).x > MAX_REPO_NAME_PIXEL_LENGTH ) {
-          name = name.substring( 0, name.length() - 1 );
-        }
-        gc.dispose();
-        name = name + "...";
-        return name;
-      }
     } );
   }
+
+  private String truncateName( String name ) {
+    GC gc = new GC( toolBar );
+    Point size = gc.textExtent( name );
+    if ( size.x <= MAX_REPO_NAME_PIXEL_LENGTH ) { // repository name is small enough to fit just return it.
+      gc.dispose();
+      return name;
+    }
+    String originalName = name;
+    while ( gc.textExtent( name + "..." ).x > MAX_REPO_NAME_PIXEL_LENGTH ) {
```

Comment:
@wseyler I know this isn't new code, but at some point it would be good to use the actual ellipsis character.